tests/end-to-end/regression/6234/Issue6234Test.php

```diff
@@ -0,0 +1,23 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\TestFixture;
+
+use PHPUnit\Framework\TestCase;
+
+final class Issue6234Test extends TestCase
+{
+    public function testExitWithoutProcessIsolation(): void
+    {
+        $this->assertTrue(true);
+        $this->assertTrue(true);
+
+        exit(1);
```

Comment:
just realized there is a even more critical case, namely a test calling `exit(0)`, from non-isolated tests.
this would make PHPUnit return a 0 exit-code and noone would notice (besides the error message), that tests did not run.

stuff for a separate issue though